### PR TITLE
T36372 Update Node model with status and result

### PR DESCRIPTION
**Pull request by JenySadadia**
Need to update Node model to accommodate status and result separately.
Update unit tests and documentation for the same.

### CODE REVIEW

File: api/models.py [hunk 1]
```diff
@@ -133,7 +139,10 @@ class Node(DatabaseModel):
     )
     status: Optional[StatusValues] = Field(
         default=StatusValues.PENDING,
-        description='Status of test result'
+        description='Status of test'
```
Comment (gctucker) [inline api/models.py:142]: It's actually not a test any more, but status of the node. This is a detail, we may not need to change it now but it'll be interesting to see how things evolve when we have actual test nodes and other kinds of nodes.

File: test/test_node_handler.py [hunk 2]
```diff
@@ -367,6 +378,7 @@ def test_get_root_node_endpoint(mock_db_find_by_id, mock_init_sub_id):
             'name',
             'parent',
             'revision',
+            'result',
```
Comment (gctucker) [inline test/test_node_handler.py:381]: I think it's a bit better to keep following the alphabetical order:
```suggestion
            'result',
            'revision',
```

**Review by gctucker — APPROVED** [review]
LGTM, just some minor things that might be tweaked.

**Comment by gctucker** [comment]
In a follow-up PR it would be good to document the state machine for the `status` field with `pending`, `pass`, `fail` and `timeout`.  Also the transitions can be enforced in the model.

File: test/test_node_handler.py (hunk 2 above)
Comment (JenySadadia) [reply test/test_node_handler.py]: Updated.

File: api/models.py (hunk 1 above)
Comment (JenySadadia) [reply api/models.py]: Updated the description for `status` and `result`.

**Comment by JenySadadia** [comment]
> In a follow-up PR it would be good to document the state machine for the `status` field with `pending`, `pass`, `fail` and `timeout`. Also the transitions can be enforced in the model.

Can you please elaborate?

**Comment by gctucker** [comment]
Sorry I mixed up status and result as they're now separate. I meant, the status can only start as `pending` and then it can only transition to `complete` or `timeout`. Then it can't go back to `pending` or change between `complete` and `timeout`. We may be able to add some code to validate these state changes and return an error if the client tries to make an "illegal" state change.

**Review by gctucker — APPROVED** [review]

**Pull request merged**